docs/pages/versions/unversioned/sdk/filesystem.md

```diff
@@ -73,6 +77,41 @@ try {
 }
 ```
 
+#### Using external assets
+
+```ts
+const getContactImageBase64 = async () => {
+  const { status } = await Contacts.requestPermissionsAsync();
+  if (status === 'granted') {
+    try {
+      const { data: contacts } = await Contacts.getContactsAsync({
+        fields: [Contacts.Fields.Image],
+      });
+
+      // find first contact with image
+      const firstContactImage = contacts.find(c => c.image != null);
+      const imageUri = firstContactImage.image.uri;
+
+      // copy image into your app's filesystem
+      const tempUri = FileSystem.cacheDirectory + 'temp_img';
+      await FileSystem.copyAsync({ from: imageUri, to: tempUri });
+
+      // get base64 string
+      const result = await FileSystem.readAsStringAsync(tempUri, {
+        encoding: FileSystem.EncodingType.Base64,
+      });
+
+      console.log(result);
+
+      // delete temporary file
+      await FileSystem.deleteAsync(tempUri);
+    } catch (e) {
+      console.error(e);
+    }
+  }
+};
+```
+
```

Comment:
I am not sure if this is good example

Reply:
I'm not 100% sure what you are demonstrating here. It feels a bit weird to copy contact images, extracting Base64 sources, and then deleting them. If this is necessary, I think we should add some helper stuff. E.g. something like this:

```
const asset = Asset.fromUri(firstContactImage.image.uri);
const string = await asset.getBase64Async();
```

Maybe highlighting how external assets can be cached, using websites or API sources might be better here?

Reply:
Thank you for your comment, it looks like you share my concerns about this example. I was thinking about showing some general usage of `FileSystem`, including copying from external assets, but I couldn't find any reasonable scenario to demonstrate. Now I'm wondering if writing another example here makes sense.

 > Maybe highlighting how external assets can be cached, using websites or API sources might be better here?

IMO this kind of stuff doesn't only fit `FileSystem`, but it covers wider set of Expo features. For instance, [this guide](https://docs.expo.io/guides/preloading-and-caching-assets/) could be a better place. By the way, referring to _some helper stuff_, I have recently discovered (#9246) that `expo-asset` already supports web URLs, but it is undocumented (notabene it uses _FileSystem_ under the hood). Anyway, I'd like to focus on `FileSystem` in this PR.

Reply:
If you do want to keep an example here, maybe showing how to cache giphys with filesystem might be a more sensible demonstration? That should contain the basics of what you have in this demonstration, but it feels a bit more real-world applicable. (e.g. adding a method to cache a list of giphys (storing external filesystem data) and a method to clear the cache of giphys (removing local filesystem data))

Reply:
I started writing the Giphy example, and [made this snack](https://snack.expo.io/@barthap10/giphy-example). It could be added as _Inline Snack_, or I can truncate it to leave only file-system related code.